### ISSUE TITLE
Some numeric values are allowed as an Enum name

### ISSUE THREAD

**Issue by a-tarasyuk** [microsoft/TypeScript#40284]
**TypeScript Version:**  4.0.2

**Search Terms:** Enum Names

**Code**

```ts
enum E1 { "0.1" }  // An enum member cannot have a numeric name.
enum E2 { "-0.1" } // An enum member cannot have a numeric name.
enum E3 { "1000" } // An enum member cannot have a numeric name.
enum E4 { "1000" } // An enum member cannot have a numeric name.
enum E5 { "0.11" } // An enum member cannot have a numeric name.

enum E6 { "+0.1" } // no error
enum E7 { "0.10" } // no error
```
**Expected behavior:**

Examples `E6` and `E7` should produce errors 

**Actual behavior:**

Examples `E6` and `E7`are valid

**Playground Link:** [**Playground**](https://www.typescriptlang.org/play?#code/KYOwrgtgBAogjFA3lARABgHRxVAvlKAekNgCdSB7UgWAChRJYAmJVAWk2zyJJnKroNoMAMysUcNFJz5iZSjXrhhAFnGTp3OXwWDlsAKzjOXWb36K9jGADZxAak4yeUEBSjALV4QHZjWNGc5Nw8LIA)

**Related Issues:** ---

**Comment by RyanCavanaugh** [comment]
This is the intended behavior; a name can't be the same as its own round-tripped numeric/string representation. The error message doesn't go into this level of precision
